lib/datasource/nuget/index.ts

```diff
@@ -15,7 +15,8 @@ export function parseRegistryUrl(
   registryUrl: string
 ): { feedUrl: string; protocolVersion: number } {
   try {
-    const parsedUrl = urlApi.parse(registryUrl);
+    // TODO: check why test is failing with whatwg url api
+    const parsedUrl = parseUrlLegacy(registryUrl);
```

Comment:
IMO the test is wrong. It asserts that `protocolVersion` is `null`, which can only happen in the catch. I am guessing `url.parse` was throwing an exception for the malformed URL, but now we're catching it internally?

If you change the code below to something like

```typescript
let protocolVersion: number;
const protocolVersionRegExp = /#protocolVersion=(2|3)/;
const protocolVersionMatch = protocolVersionRegExp.exec(parsedUrl.hash);
if (protocolVersionMatch) {
  parsedUrl.hash = '';
  protocolVersion = Number.parseInt(protocolVersionMatch[1], 10);
} else if (parsedUrl.pathname.endsWith('.json')) {
  protocolVersion = 3;
} else {
  protocolVersion = 2
}
```

that might work?